cli/cli/helpers/grafloki/shared.go

```diff
@@ -31,3 +39,59 @@ type GrafanaDatasources struct {
 	ApiVersion  int64               `yaml:"apiVersion"`
 	Datasources []GrafanaDatasource `yaml:"datasources"`
 }
+
+func StartGrafloki(ctx context.Context, clusterType resolved_config.KurtosisClusterType, graflokiConfig resolved_config.GrafanaLoki) (logs_aggregator.Sinks, string, error) {
+	var lokiHost string
+	var grafanaUrl string
+	var err error
+	switch clusterType {
+	case resolved_config.KurtosisClusterType_Docker:
+		lokiHost, grafanaUrl, err = StartGrafLokiInDocker(ctx, graflokiConfig)
+		if err != nil {
+			return nil, "", stacktrace.Propagate(err, "An error occurred starting Grafana and Loki in Docker.")
+		}
+	case resolved_config.KurtosisClusterType_Kubernetes:
+		lokiHost, grafanaUrl, err = StartGrafLokiInKubernetes(ctx, graflokiConfig)
+		if err != nil {
+			return nil, "", stacktrace.Propagate(err, "An error occurred starting Grafana and Loki in Kubernetes.")
+		}
+	default:
+		return nil, "", stacktrace.NewError("Unsupported cluster type: %v", clusterType.String())
+	}
+
+	// This matches the exact configurations here: https://vector.dev/docs/reference/configuration/sinks/loki/
+	lokiSink := map[string]map[string]interface{}{
+		"loki": {
+			"type":     "loki",
+			"endpoint": lokiHost,
+			"encoding": map[string]string{
+				"codec": "json",
+			},
+			"labels": map[string]string{
+				"job": "kurtosis",
+			},
+		},
+	}
+
+	return lokiSink, grafanaUrl, nil
+}
+
+func StopGrafloki(ctx context.Context, clusterType resolved_config.KurtosisClusterType) error {
+	switch clusterType {
+	case resolved_config.KurtosisClusterType_Docker:
+		err := StopGrafLokiInDocker(ctx)
+		if err != nil {
+			return stacktrace.Propagate(err, "An error occurred stopping Grafana and Loki containers in Docker.")
+		}
+	case resolved_config.KurtosisClusterType_Kubernetes:
+		err := StopGrafLokiInKubernetes(ctx)
+		if err != nil {
+			return stacktrace.Propagate(err, "An error occurred stopping Grafana and Loki containers in Kubernetes.")
+		}
+	default:
+		return stacktrace.NewError("Unsupported cluster type: %v", clusterType.String())
+	}
+
+	out.PrintOutLn("Successfully stopped Grafana and Loki containers.")
```

Comment:
Still confused about why out.PrintOutLn vs logrus. Seems like should be `logrus` always, so that we can capture CLI output by redirecting logrus' output source (if the CLI is using `out.PrintOutLn` then it's printing directly to STDOUT and tests can't capture and inspect that)